Testcase "test_environment_variables" under Configuration (opentelemery-api)  is failing with AssertionError

**Describe your environment** 

Environment:
OS: Windows
Python version : 3.8.2 (64bit)

**Steps to reproduce**
1. Follow the [contributing guidelines](https://github.com/open-telemetry/opentelemetry-python/blob/master/CONTRIBUTING.md) to install the opentelemetry-python repo in development mode.
2. Then run the tests using tox -e test-core-api

**What is the expected behavior?**
All the testcases should PASS.

What did you expect to see?
A success message conveying all testcases have passed

**What is the actual behavior?**

The testcase "test_environment_variables" under Configuration fails with the error :
```
 def test_environment_variables(self) -> None:
        self.assertEqual(Configuration().METER_PROVIDER, "meter_provider")  # pylint: disable=no-member
        self.assertEqual( Configuration().TRACER_PROVIDER, "tracer_provider" )  # pylint: disable=no-member
      self.assertEqual( Configuration().OThER, "other")  # pylint: disable=no-member 
```

`       AssertionError: None != 'other' `
 `opentelemetry-api\tests\configuration\test_configuration.py:50: AssertionError `

`The statement self.assertEqual(Configuration().OThER, "other" )  fails `

What did you see instead?
One test case failed in the whole set.

**Additional context**
Reason for failure:

- The test patches os.environ by adding environment variable "OTEL_OThER"  in the form os.environ["OTEL_OThER"] = "other"
- In windows, os.py module converts the variable to uppercase before storing. 
- Hence variable get stored as **OTEL_OTHER**

But the testcase tries to access Configuration().OThER , which does not exist.

Observation :
Changing the testcase by modifying the assertEqual(Configuration().OTHER, "other') makes the testcase PASS
Accessing **os.environ["OTEL_OThER"]**   OR  **os.environ["OTEL_OTHER"]** ***does not*** throw any key error.

--- a/opentelemetry-api/tests/configuration/test_configuration.py
+++ b/opentelemetry-api/tests/configuration/test_configuration.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 # pylint: disable-all
 
+from sys import platform
 from unittest import TestCase
 from unittest.mock import patch
 
@@ -35,7 +36,7 @@ class TestConfiguration(TestCase):
         {
             "OTEL_PYTHON_METER_PROVIDER": "meter_provider",
             "OTEL_PYTHON_TRACER_PROVIDER": "tracer_provider",
-            "OTEL_OThER": "other",
+            "OTEL_OTHER" if platform == "windows" else "OTEL_OThER": "other",
             "OTEL_OTHER_7": "other_7",
             "OPENTELEMETRY_PTHON_TRACEX_PROVIDER": "tracex_provider",
         },

